rest-api/README.md

```diff
@@ -0,0 +1,321 @@
+
+### Query Params
+| Param              | Description                                                                                                                                                        |
+| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `baseCurrency`     | The primary currency listed in a currency pair, representing the unit being traded. **This is the currency that Alice will be _receiving_**                        |
+| `quoteCurrency`    | The secondary currency listed in a currency pair, indicating its value in relation to the base currency. **This is the currency that the PFI will be _receiving_** |
+| `payinMethodKind`  | The payin method Alice wishes to use to provide quote currency                                                                                                     |
+| `payoutMethodKind` | The payout method Alice wishes to use to receive base currency                                                                                                     |
+| `id`               | Query for a specific offering                                                                                                                                      |
+
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `200: OK`          | `Offering[]`          |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+---
+
+# Threads
+A thread is a series of linked tbDEX messages between Alice and a PFI for a single exchange. A thread can be created by submitting an RFQ.
+
+## Submit RFQ
+
+### Endpoint
+`POST /threads/:thread_id/rfq`
+
+### Authentication
+Refer to [Signature Verification Section]() of the tbDEX spec  
+### Authorization
+No Authorization required to submit an RFQ
+
+### Request Body
+> [!IMPORTANT]
+> See RFQ structure [here](../README.md#rfq-request-for-quote)
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description            |
+| ------ | ---------------------- |
+| 400    | Validation error(s)    |
+| 400    | Failed Signature Check |
+| 404    | Thread not found       |
+| 409    | RFQ already exists     |
+| 409    | Thread already exists  |
+
+## Get Quote
+### Endpoint
+`GET /threads/:thread_id/?messageType=quote`
+
+### Response
+| Status             | Body                              |
+| ------------------ | --------------------------------- |
+| `200: OK`          | `{ data: TbdexMessage<Quote>[] }` |
+| `400: Bad Request` | `{ errors: Error[] }`             |
+
+### Response Body
+> [!IMPORTANT]
+> See Quote structure [here](../README.md#quote)
+
+## Submit Order
+
+### Endpoint
+`POST /threads/:thread_id/order`
```

Comment:
Resurfacing a proposal from [here](https://sq-tbd.slack.com/archives/C05B6AB7JG1/p1692319359593849?thread_ts=1692307921.943449&cid=C05B6AB7JG1)

For both _order_ and _close_...

```
PUT /threads/:id

{
  "messageType": "order|close"
}
```

...and also... I would open the question of `action` rather than `messageType` (oh, or `command` per 
@phoebe-lew's comment)

----

The big thing is, no need for a `parentId` in either request. Because a `parentId` means that the client would either need to manage the state of the prior message in the thread, or else first make a call to `GET /threads/:id` as a means to retrieve the `parentId`.

The exact RESTful spec is neither here nor there, like `POST /threads/:id/order` has the same inputs as `PUT /threads/:id`+request body. But the `PUT /threads/:id` means you're exposing a way to _mutate the thread_ which is semantically different than the `POST` design which means _create an order/close_

Reply:
will open an issue and surface this